Vacuum `nil` pointer dereference

There is a race condition in which if the Broker exists but it has not been processed by Knative Eventing, the Vacuum raises a `nil` pointer dereference.

This is the error:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1e2a54a]

goroutine 1 [running]:
github.com/kubearchive/kubearchive/pkg/cloudevents.getBrokerUrl()
        github.com/kubearchive/kubearchive/pkg/cloudevents/sink_publisher.go:227 +0x1ea
github.com/kubearchive/kubearchive/pkg/cloudevents.NewSinkCloudEventPublisher({0x2506385, 0x13}, {0x251778e, 0x1d})
        github.com/kubearchive/kubearchive/pkg/cloudevents/sink_publisher.go:54 +0x1a5
main.clusterVacuum({0x7ffe4225882b, 0x11})
        github.com/kubearchive/kubearchive/cmd/vacuum/cluster_vacuum.go:27 +0xa6
main.main()
        github.com/kubearchive/kubearchive/cmd/vacuum/main.go:20 +0xfb
```

These are the brokers:

```
NAME                 URL   AGE    READY   REASON
kubearchive-broker         124m           
kubearchive-dls            124m           
```

To reproduce it I reordered and commented some parts of the `hach/quick-install.sh` script, specifically after installing Knative Eventing I'm not waiting for it to be responsive:

```
#!/bin/bash
# Copyright KubeArchive Authors
# SPDX-License-Identifier: Apache-2.0

set -o errexit
set -o xtrace

SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
cd ${SCRIPT_DIR}/..

bash integrations/database/postgresql/install.sh

# renovate: datasource=github-releases depName=cert-manager packageName=cert-manager/cert-manager
export CERT_MANAGER_VERSION=v1.17.2
# renovate: datasource=github-releases depName=knative-eventing packageName=knative/eventing
export KNATIVE_EVENTING_VERSION=v1.18.1

kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/${CERT_MANAGER_VERSION}/cert-manager.yaml
kubectl rollout status deployment --namespace=cert-manager --timeout=30s

kubectl apply -f https://github.com/knative/eventing/releases/download/knative-${KNATIVE_EVENTING_VERSION}/eventing.yaml
# kubectl rollout status deployment --namespace=knative-eventing --timeout=50s

bash hack/kubearchive-install.sh
```

**Acceptance Criteria**

* Handle this case and yield a more useful error, like "Broker does not have a URL", or something like that. Maybe we can check if the state is ready.

--- a/pkg/cloudevents/sink_publisher.go
+++ b/pkg/cloudevents/sink_publisher.go
@@ -5,6 +5,7 @@ package cloudevents
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"log/slog"
 	"strings"
@@ -233,6 +234,11 @@ func getBrokerUrl() (string, error) {
 	if err != nil {
 		slog.Error("Failed to get KubeArchive broker", "error", err)
 		return "", err
+	}
+
+	if broker.Status.Address == nil {
+		slog.Error("KubeArchive broker has no address assigned")
+		return "", errors.New("broker does not have an address assigned")
 	}
 
 	return broker.Status.Address.URL.String(), nil

